src/Components/WebAssembly/testassets/ThreadingApp/Pages/Counter.razor

```diff
@@ -0,0 +1,79 @@
+﻿@page "/counter"
+@using System.Runtime.InteropServices
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        if(!RuntimeInformation.IsOSPlatform(OSPlatform.Create("Browser")))
+        {
+            return;
+        }
+
+        if (Thread.CurrentThread.ManagedThreadId != 1)
+        {
+            throw new Exception("We should be on main thread!");
+        }
+
+        Exception exc = null;
+        try
+        {
+            // send me to the thread pool
+            await Task.Delay(10).ConfigureAwait(false);
```

Comment:
Doesn't `Task.Yield` achieve the same effect?

Reply:
probably yes but `Task.Yield` doesn't have `ConfigureAwait`. Also I think this will force the scheduler to not invoke it synchronously.

Reply:
You could now do this: `await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);`
https://github.com/dotnet/runtime/pull/87067#discussion_r1215280841